model/src/main/proto/survey.proto

```diff
@@ -0,0 +1,204 @@
+syntax = "proto3";
+
+package model;
+
+option java_package = "org.oppia.android.app.model";
+option java_multiple_files = true;
+
+// Corresponds to a single survey.
+message Survey {
+  // The ID of the survey. This is a string value that uniquely identifies the survey.
+  // The ID is guaranteed to be unique across all surveys globally for the purpose of analysis.
+  // This field is used to retrieve information about a survey, and should be treated as read-only.
+  string survey_id = 1;
+
+  // A list of questions that make up the survey.
+  repeated SurveyQuestion questions = 2;
+}
+
+// Represents a question that is part of the Survey.
+message SurveyQuestion {
+  // The ID of the survey question. This is a string value that uniquely identifies the question
+  // within the context of the survey, and is guaranteed to be unique within the survey scope only.
+  string question_id = 1;
+
+  // The semantic name of the question used to identify it in the analytics dashboard.
+  QuestionName question_name = 2;
+
+  // The actual text value of the question.
+  string question = 3;
+
+  // The type of question.
+  oneof question_type {
+    // Corresponds to a question with multiple choices.
+    SurveyQuestionOptionList multiple_choice = 4;
+
+    // Corresponds to a question with multiple choices and a free-text 'other' option.
+    SurveyQuestionOptionList multiple_choice_with_other = 5;
+
+    // Indicates whether this question takes a free-text response.
+    bool free_form_text = 6;
+  }
+
+  // The language code to which this question is localized.
+  string language_code = 7;
+}
+
+// Represents a list of answer options for a multiple choice question.
+message SurveyQuestionOptionList {
+  repeated SurveyQuestionOption options = 1;
+}
+
+// Corresponds to the semantic name given to each question.
+enum QuestionName {
```

Comment:
Should this be ``SurveyQuestionName`` for consistency elsewhere in this file? It seems like everything tends to start with 'survey'.

Reply:
> It seems like everything tends to start with 'survey'.

This is only because some of the message names have been used in other messages. 

I have updated this message to start with survey too, for consistency.